datafusion/core/src/physical_plan/aggregates/mod.rs

```diff
@@ -1506,24 +1526,53 @@ mod tests {
         let result =
             common::collect(partial_aggregate.execute(0, task_ctx.clone())?).await?;
 
-        let expected = vec![
-            "+---+-----+-----------------+",
-            "| a | b   | COUNT(1)[count] |",
-            "+---+-----+-----------------+",
-            "|   | 1.0 | 2               |",
-            "|   | 2.0 | 2               |",
-            "|   | 3.0 | 2               |",
-            "|   | 4.0 | 2               |",
-            "| 2 |     | 2               |",
-            "| 2 | 1.0 | 2               |",
-            "| 3 |     | 3               |",
-            "| 3 | 2.0 | 2               |",
-            "| 3 | 3.0 | 1               |",
-            "| 4 |     | 3               |",
-            "| 4 | 3.0 | 1               |",
-            "| 4 | 4.0 | 2               |",
-            "+---+-----+-----------------+",
-        ];
+        let expected = if spill {
+            vec![
+                "+---+-----+-----------------+",
```

Comment:
@kazuyukitanimura Hi, do you remember why the result of `spill` is different?

Reply:
I think the reason is that we are comparing the output of a partial aggregate and when `spill` we also have a lower desired batch size and hit the emit early logic: https://github.com/apache/datafusion/blob/5b6b404b9fbccb8584e8c5d2f1c366f3de736f31/datafusion/physical-plan/src/aggregates/row_hash.rs#L934-L945

Reply:
That's right. When spilling happens, it means we don't have enough memory to hold original batch, different partial aggregation result will be but I think it doesn't change final aggregation result.